File: src/main/java/org/funivan/lologa/tiles/TilesInterface.java
package org.funivan.lologa.tiles;

import org.funivan.lologa.tile.Position.PositionInterface;
import org.funivan.lologa.tile.TileInterface;

public interface TilesInterface {

    TileInterface get(PositionInterface position);

    boolean has(PositionInterface position);

    TilesInterface without(PositionInterface position);

    TilesInterface with(TileInterface tile);

    Iterable<TileInterface> all();


}

File: src/main/java/org/funivan/lologa/tiles/TilesInterface.kt
package org.funivan.lologa.tiles

import org.funivan.lologa.tile.Position.PositionInterface
import org.funivan.lologa.tile.TileInterface

interface TilesInterface {

    operator fun get(position: PositionInterface): TileInterface

    fun has(position: PositionInterface): Boolean

    fun without(position: PositionInterface): TilesInterface

    fun with(tile: TileInterface): TilesInterface

    fun all(): Iterable<TileInterface>

    fun size(): Int


}
